addons/sale/models/sale_order_line.py

```diff
@@ -607,7 +620,7 @@ def _convert_to_tax_base_line_dict(self, **kwargs):
             taxes=self.tax_id,
             price_unit=self.price_unit,
             quantity=self.product_uom_qty,
-            discount=self.discount,
+            discount=self._get_real_discount(self.discount),
```

Comment:
There are different places where we use the amount to compute taxes, I guess it should be used there too.